### PR TITLE
[DPE-4987] Fix failing tests on latest edge using secrets

### PR DESCRIPTION
**Pull request by deusebio**
Addressing issue #94 

* The changes have been tested against main (without secrets) in [this workflow](https://github.com/canonical/kafka-bundle/actions/runs/10169045373/job/28125067554)
* The changes have been tested against edge (with secrets) in [this workflow](https://github.com/canonical/kafka-bundle/actions/runs/10170282741/job/28129148579).

### CODE REVIEW

File: tests/integration/bundle/kafka_helpers.py [hunk 1]
```diff
@@ -81,25 +88,132 @@ def show_unit(unit_name: str, model_full_name: str) -> Any:
     return yaml.safe_load(result)
 
 
-def get_zookeeper_connection(unit_name: str, model_full_name: str) -> Tuple[List[str], str]:
-    result = show_unit(unit_name=unit_name, model_full_name=model_full_name)
+def get_secret_by_label(model_full_name: str, label: str, owner: str) -> dict[str, str]:
```
Comment (deusebio) [inline tests/integration/bundle/kafka_helpers.py:91]: I have updated this implementation, since in the [previous implementation](https://github.com/canonical/kafka-operator/blob/main/tests/integration/helpers.py#L310) I found an edge case that when there was no secret matching with owner and label, the data returned was the last `secret_id` (given the previous for loop-continue-break pattern - which I found it a bit of a weird pattern TBH)

**Review by marcoppenheimer — APPROVED** [review]
Looks good!
Weird secret behavior though, glad you found a fix.

**Review by zmraul — APPROVED** [review]
